File: app/src/main/java/com/pr0gramm/app/ui/FeedItemViewHolder.java
package com.pr0gramm.app.ui;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;

import com.pr0gramm.app.R;

import static butterknife.ButterKnife.findById;
import static com.google.common.base.Preconditions.checkNotNull;

/**
 * View holder for one feed item.
 */
public final class FeedItemViewHolder extends RecyclerView.ViewHolder {
    public final ImageView seen;
    public final ImageView repost;
    public final View preloaded;
    public final ImageView image;

    public FeedItemViewHolder(View itemView) {
        super(itemView);

        image = checkNotNull(findById(itemView, R.id.image));
        seen = checkNotNull(findById(itemView, R.id.seen));
        repost = checkNotNull(findById(itemView, R.id.repost));
        preloaded = checkNotNull(findById(itemView, R.id.preloaded));
    }

    public void setIsRepost() {
        repost.setVisibility(View.VISIBLE);
        seen.setVisibility(View.GONE);
    }

    public void setIsSeen() {
        seen.setVisibility(View.VISIBLE);
        repost.setVisibility(View.GONE);
    }

    public void clear() {
        seen.setVisibility(View.GONE);
        repost.setVisibility(View.GONE);
    }

    public void setIsPreloaded(boolean isPreloaded) {
        preloaded.setVisibility(isPreloaded ? View.VISIBLE : View.GONE);
    }
}

File: app/src/main/java/com/pr0gramm/app/ui/FeedItemViewHolder.kt
package com.pr0gramm.app.ui

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import com.pr0gramm.app.R
import com.pr0gramm.app.feed.FeedItem
import com.pr0gramm.app.util.find
import com.pr0gramm.app.util.visible

/**
 * View holder for one feed item.
 */
class FeedItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private val seen: ImageView = find(R.id.seen)
    private val repost: ImageView = find(R.id.repost)
    private val preloaded: View = find(R.id.preloaded)

    val image: ImageView = find(R.id.image)

    var item: FeedItem? = null
    var index: Int = 0

    fun setIsRepost() {
        repost.visible = true
        seen.visible = false
    }

    fun setIsSeen() {
        seen.visible = true
        repost.visible = false
    }

    fun setIsPreloaded(isPreloaded: Boolean) {
        preloaded.visible = isPreloaded
    }

    fun clear() {
        seen.visible = false
        repost.visible = false
    }
}
